hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -678,6 +695,26 @@ private void insertTokenTransfers(RecordItem recordItem) {
         }
     }
 
+    private void insertAutomaticTokenAssociations(RecordItem recordItem) {
```

Comment:
I think it would be better to have this method applicable to both. So there's a single central place for automatic token associations.

Maybe update this method to be
```suggestion
    private void insertAutomaticTokenAssociations(TokenAssociationsList tokenAssociationsList, TokenFreezeStatusEnum freezeStatus, TokenKycStatusEnum kycStatus) {
```

For create you can pass the status in, for others it will be null

Reply:
this doesn't really help.

the automatic token associations of a token create transaction is handled in `insertTokenCreate` because

- the freeze status & kyc status are set differently
- there is backwards compatibility that before 0.18.0 there is no automatic_token_associations in transaction record

The best if we want is re-parse the tokencreate transaction for its freeze key & kyc key in `insertAutomaticTokenAssociations`, still in `insertTokenCreate` we need to skip the prior-to-0.18.0 logic if the list is not empty